portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,9 +25,23 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  
+  private String output;
+  
+  public DataServlet () {
+   String output = "No Input Received.";
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    response.getWriter().println(output);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    output = new Gson().toJson(request.getParameter("input"));
```

Comment:
FYI: in follow-on steps you may want to use datastore to persist the data in the request. For now this data is only cached on the server.